src/gt4py/stencil_object.py

```diff
@@ -0,0 +1,294 @@
+                by doing: `{name: value for name, value in kwargs.items()}`
+
+            domain : `Sequence` of `int`, optional
+                Shape of the computation domain. If `None`, it will be used the
+                largest feasible domain according to the provided input fields
+                and origin values (`None` by default).
+
+            origin :  `[int * ndims]` or `{'field_name': [int * ndims]}`, optional
+                If a single offset is passed, it will be used for all fields.
+                If a `dict` is passed, there could be an entry for each field.
+                A special key *'_all_'* will represent the value to be used for all
+                the fields not explicitly defined. If `None` is passed or it is
+                not possible to assign a value to some field according to the
+                previous rule, the value will be inferred from the global boundaries
+                of the field. Note that the function checks if the origin values
+                are at least equal to the `global_border` attribute of that field,
+                so a 0-based origin will only be acceptable for fields with
+                a 0-area support region.
+
+            exec_info : `dict`, optional
+                Dictionary used to store information about the stencil execution.
+                (`None` by default).
+
+        Returns
+        -------
+            `None`
+
+        Raises
+        -------
+            ValueError
+                If invalid data or inconsistent options are specified.
+        """
+
+        if exec_info is not None:
+            exec_info["call_run_start_time"] = time.perf_counter()
+        used_arg_fields = {
+            name: field
+            for name, field in field_args.items()
+            if name in self.field_info and self.field_info[name] is not None
+        }
+        used_arg_params = {
+            name: param
+            for name, param in parameter_args.items()
+            if name in self.parameter_info and self.parameter_info[name] is not None
+        }
+        for name, field_info in self.field_info.items():
+            if field_info is not None and field_args[name] is None:
+                raise ValueError("Field '{field_name}' is None.".format(field_name=name))
+        for name, parameter_info in self.parameter_info.items():
+            if parameter_info is not None and parameter_args[name] is None:
+                raise ValueError(
+                    "Parameter '{parameter_name}' is None.".format(parameter_name=name)
+                )
+        # assert compatibility of fields with stencil
+        for name, field in used_arg_fields.items():
+            if not gt_backend.from_name(self.backend).storage_info["is_compatible_layout"](field):
+                raise ValueError(
+                    "The layout of the field {} is not compatible with the backend.".format(name)
```

Comment:
I'd rather use f-strings than .format() without field names; it is much easier to read: 

`f"The layout of the field {name} is not compatible with the backend."`

Reply:
It looks like done